staging/src/k8s.io/client-go/plugin/pkg/client/auth/oidc/oidc.go

```diff
@@ -120,7 +120,10 @@ func newOIDCAuthProvider(_ string, cfg map[string]string, persister restclient.A
 
 	// Check cache for existing provider.
 	if provider, ok := cache.getClient(issuer, clientID); ok {
-		return provider, nil
+		useCache := provider.updateTokenAndCheckMatch(cfg)
```

Comment:
This cache was not designed for multi-user use (which is indistinguishable from a single-user use where the token changes). Rather than continuing to only cache a single provider per client id, this may need rework to cache by host (first param to newOIDCAuthProvider),issuer,clientID,token

Reply:
Thanks for the review. The cache was not designed for mutli-user, so in this PR, if a new user comes, the old one will be removed.

But the first param to newOIDCAuthProvider is the address of api-server, see [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/rest/plugin.go#L65). I don't understand what we benefit if including cluster address as a key of the cache.

Correct me if I am wrong.
```go
type cacheKey struct {
        clusterAddress string  // <- add cluster address as part of cacheKey?
	issuerURL string
	clientID  string
}
```

Reply:
you're right, I think just {issuerURL,clientID,token} is the key we care about

Reply:
I think the `token` (or `cfg[cfgRefreshToken]` to be precise) can not be the caching key here. From the issue description (https://github.com/kubernetes/kubernetes/issues/78494), the refresh token can be updated multiple times over the lifetime of the caller. If we included `cfg[cfgRefreshToken]` as a cache key, the amount of entries in the cache could grow and we'd probably need to introduce some cache invalidation mechanism. 

However, I think we could introduce a middle ground by leaving the cache key the same but also checking if the `cfg[cfgRefreshToken]` has changed. This also required deep copy of the `cfg` when creating a client but this shouldn't be a problem.